### DISCUSSION TITLE
Flows retry and pool

### DISCUSSION THREAD

**Discussion by DeanKamali** [directus/directus#13815] [Archive]
Hello 

I'm wondering if this is possible with the current implementation:

- Is it possible to define retry logic if a webhook fails? 
Example: endpoint return unexpected response "error status code, or error message in response body"  

- If we were to use Flows to call 3rd party API, is it possible to pool status (pending / success) before proceeding to other Flows logic. 

- When would Flows make it to your cloud offering? 

Thank you

**Comment by benhaynes** [comment]
Great questions/ideas (love the idea around retries and reacting to responses)... I'll let the engineers answer the first two, but can say that Flows will be moving up to Cloud either today or in next Thursday's Cloud release.

**Comment by rijkvanzanten** [comment]
> Is it possible to define retry logic if a webhook fails?

Yes, every operation has a resolve and a reject handler. You can use the reject handler of the request operation to respond to failed requests.

> If we were to use Flows to call 3rd party API, is it possible to pool status (pending / success) before proceeding to other Flows logic.

The request operation is blocking by default, so it will wait while the request is pending.

> When would Flows make it to your cloud offering?

Very soon!
